addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, models, _
+from odoo.tools import config, float_round
+
+class ReportBomStructure(models.Model):
+    _name = 'report.mrp.report_bom_structure'
+
+    @api.model
+    def get_report_values(self, docids, data=None):
+        docs = []
+        for bom_id in docids:
+            bom = self.env['mrp.bom'].browse(bom_id)
+            for product_variant_id in bom.product_tmpl_id.product_variant_ids:
+                doc = self._get_pdf_line(bom_id, product_id=product_variant_id, unfolded=True)
+                doc['report_type'] = 'pdf'
+                doc['report_structure'] = 'all'
+                docs.append(doc)
+        return {
+            'doc_ids': docids,
+            'doc_model': 'mrp.bom',
+            'docs': docs,
+        }
+
+    @api.model
+    def get_html(self, bom_id=False, searchQty=1, searchVariant=False):
+        res = self._get_report_data(bom_id=bom_id, searchQty=searchQty, searchVariant=searchVariant)
+        res['lines']['report_type'] = 'html'
+        res['lines']['report_structure'] = 'all'
+        res['lines'] = self.env.ref('mrp.report_mrp_bom').render({'data': res['lines']})
+        return res
+
+    @api.model
+    def get_bom(self, bom_id=False, product_id=False, line_qty=False, line_id=False, level=False):
+        lines = self._get_bom(bom_id=bom_id, product_id=product_id, line_qty=line_qty, line_id=line_id, level=level)
+        return self.env.ref('mrp.report_mrp_bom_line').render({'data': lines})
+
+    @api.model
+    def get_operations(self, bom_id=False, qty=0, level=0):
+        bom = self.env['mrp.bom'].browse(bom_id)
+        lines = self._get_operation_line(bom.routing_id, qty, level)
+        values = {
+            'bom_id': bom_id,
+            'currency': self.env.user.company_id.currency_id,
+            'operations': lines,
+        }
+        return self.env.ref('mrp.report_mrp_operation_line').render({'data': values})
+
+    @api.model
+    def get_pdf(self, child_bom_ids=[], searchQty=1, searchVariant=False, report_structure='all'):
```

Comment:
you shouldn't override this method :(

Reply:
actually it isn't overriden, get_pdf is a <v10 method on report
should it be public though?